src/common/components/repositories-home/index.js

```diff
@@ -19,16 +20,19 @@ const SNAP_POLL_PERIOD = (15 * 1000);
 
 class RepositoriesHome extends Component {
   fetchData(props) {
-    const { snaps } = props;
+    const { hasSnaps, snaps, entities } = props;
 
-    if (snaps.success) {
+    // check both success and fetching to avoid redirecting
+    // based on previous success
+    if (snaps.success && !snaps.isFetching) {
```

Comment:
"Success is the resolution of a Promise, not a matter of state." -- Abraham Lincoln